numpy/f2py/crackfortran.py

```diff
@@ -2735,19 +2774,23 @@ def analyzevars(block):
                     check = None
             if dim and 'dimension' not in vars[n]:
                 vars[n]['dimension'] = []
-                for d in rmbadname([x.strip() for x in markoutercomma(dim).split('@,@')]):
+                for d in rmbadname(
+                        [x.strip() for x in markoutercomma(dim).split('@,@')]
+                ):
+                    # d is the expression inside the dimension declaration
                     star = '*'
                     if d == ':':
                         star = ':'
```

Comment:
I'm not sure I understand the variable naming scheme here; `star = ':'` seems a bit strange.

Reply:
It might be nicer to directly assign `dl=[:]` here.

Reply:
Also this should be moved below the `try...except` block; since `param_parse` can also return `:`.

Reply:
> It might be nicer to directly assign dl=[:] here.

Because `d` might be different from either `*` or `:`, this wouldn't be as direct. I'll change the name of this variable but I'd prefer leaving the logic as it is for now and maybe fixing it in a follow-up, if that's ok.